master/internal/trials/postgres_trials.go

```diff
@@ -352,12 +352,37 @@ func MetricsTimeSeries(trialID int32, startTime time.Time,
 	subq := db.Bun().NewSelect().TableExpr(tableName).
 		ColumnExpr("(select setseed(1)) as _seed").
 		ColumnExpr("total_batches as batches").
-		ColumnExpr("trial_id").ColumnExpr("end_time as time").
-		ColumnExpr("(metrics ->'?' ->> 'epoch')::float8 as epoch", bun.Safe(metricsObjectName))
+		ColumnExpr("trial_id").ColumnExpr("end_time as time")
 
-	for _, metricName := range metricNames {
-		subq = subq.ColumnExpr("(metrics ->'?' ->> ?)::float8 as ?",
-			bun.Safe(metricsObjectName), metricName, bun.Ident(metricName))
+	type summary struct {
+		bun.BaseModel `bun:"table:trials"`
+		Metrics       map[string]any
+	}
+	var summaryMetrics summary
+	if err := db.Bun().NewSelect().Table("trials").
+		ColumnExpr("summary_metrics->? AS metrics", metricsObjectName).
+		Where("id = ?", trialID).
+		Scan(context.TODO(), &summaryMetrics); err != nil {
+		return nil, fmt.Errorf("getting summary metrics for trial %d: %w", trialID, err)
+	}
```

Comment:
Leaving off the cast as is will return every column as a string

If we instead do something like `metrics -> 'validation_metrics'` we will get a correctly typed map back for the field. I think we can't do this since we need the metric names as fields on the result row to apply the "PolymorphicFilter".

I think also we will need a cast anyway for "PolymorhicFilter". Do you have any ideas because I agree that this is somewhat not ideal having to lookup the datatype.

Reply:
what do you think about
1. Construct jsonb_build_object as columnexpr, e.g. `subq = subq.ColumnExpr("jsonb_build_object('loss', metrics->'avg_metrics'->'loss', 'loss2', metrics->'avg_metrics'->'loss2') as values_map")` but generate the `jsonb_build_object` parameters from `metricNames`.
2. it'll arrive as a json string we can parse into `map[string]interface{}`:
```
        var valuesMap map[string]interface{}
		if res, ok := results[i]["values_map"].(string); ok {
			err := json.Unmarshal([]byte(res), &valuesMap)
			if err != nil {
				return nil, errors.Wrap(err, "!!!!!!!!!! NOT OK")
			}
		} else {
			return nil, fmt.Errorf("UH-OH")
		}

		metricM := db.MetricMeasurements{
            ....
			Values:  valuesMap,
		}
```

Reply:
We still need the metrics on the row because of ```ApplyPolymorphicFilter```  that allows filtering on these values?

I think even if we pass in the json path instead of the column to build the filter we would still need to add type casts and would take some refactoring to be able to pass the path in safely

Reply:
PolymorphicFilter parameter already has the type parameter built in, we could do a cast for for it.

yes, query escaping would be messy. okay, let's merge this, the alternative path is hardly any better.